### DISCUSSION TITLE
Reading CSV files with comma as decimal separator, without set the field type

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#13295] [Q&A] [answered]
Hi,
I have this input CSV file, in which the decimal separator is the `,` (it is the separator in Italy): https://gist.github.com/aborruso/79dcde019e4771e64a30db23e4e0d4aa

If I read it in this way

```
duckdb -c "SUMMARIZE SELECT * from read_csv('https://gist.githubusercontent.com/aborruso/79dcde019e4771e64a30db23e4e0d4aa/raw/50f73812621bae24d063bac15ebb3b1701ef0cc5/tmp.csv',normalize_names=true,decimal_separator=',',sample_size=-1,header=true, auto_detect=1)"
```

I have in output only varchar fields.

Reading [this response](https://github.com/duckdb/duckdb/discussions/9363#discussioncomment-7298419), I had understood that auto inferencing was done, and that the value "10999999,46" was read as a decimal.

Instead, the only way to have those columns as decimal fields is to write the field type for each column (`columns={'name1': 'varchar', 'name2': 'float'}`). 
Is it correct? Is there no way to use the "decimal_separator" parameter, without doing the hard coding of the field type for each numeric field?

Thank you

**Comment by pdet** [comment]
Hi @aborruso, the sniffer does not automatically sniff decimal types (see https://duckdb.org/2023/10/27/csv-sniffer.html#type-detection)
The decimal separator is used only for double/float values in the the case of the default sniffer.

If you want to set the decimal types you indeed must either hard code them, or use the [`auto_type_candidates` option](https://duckdb.org/docs/data/csv/overview#auto_type_candidates-details).

**Reply by aborruso** [reply]
Thank you very much @pdet .

If I add `auto_type_candidates`,nothing changes, all fields are of type "VARCHAR".

Am I misusing the command or am I actually forced to do hard coding of the field type for all the double fields in my example?

Thank you again


```
duckdb --csv -c "SUMMARIZE SELECT * from read_csv('https://gist.githubusercontent.com/aborruso/79dcde019e4771e64a30db23e4e0d4aa/raw/50f73812621bae24d063bac15ebb3b1701ef0cc5/tmp.csv',normalize_names=true,decimal_separator=',',sample_size=-1,header=true, auto_type_candidates = ['BIGINT', 'DOUBLE'])"
```

**Comment by pdet** [comment]
Ah, I think I misunderstood your question, so even though you set `decimal_separator=',',sample_size=-1` it still does not identify the columns with numeric values and the `,` as doubles. Is that it?

I just ran it on `main`,  this is the result I got:

```sql
select columns FROM sniff_csv('/Users/holanda/Desktop/79dcde019e4771e64a30db23e4e0d4aa-50f73812621bae24d063bac15ebb3b1701ef0cc5/tmp.csv',decimal_separator=',',sample_size=-1);
```

```
[{'name': CUP, 'type': VARCHAR}, {'name': Codice Locale Progetto, 'type': VARCHAR}, {'name': Finanziamento Prosecuzione Opere Pubbliche - FPOP, 'type': DOUBLE}, {'name': Finanziamento UE (Diverso da PNRR), 'type': DOUBLE}, {'name': Finanziamento Regione, 'type': DOUBLE}, {'name': Finanziamento Provincia, 'type': DOUBLE}, {'name': Finanziamento Comune, 'type': DOUBLE}, {'name': Finanziamento Altro Pubblico, 'type': DOUBLE}, {'name': Finanziamento Privato, 'type': DOUBLE}, {'name': Finanziamento Da Reperire, 'type': DOUBLE}, {'name': Finanziamento PNRR, 'type': DOUBLE}, {'name': Finanziamento PNC, 'type': DOUBLE}, {'name': Altri Fondi, 'type': DOUBLE}, {'name': Finanziamento Totale, 'type': DOUBLE}, {'name': Finanziamento Totale Pubblico, 'type': DOUBLE}, {'name': Finanziamento Totale Pubblico Netto, 'type': DOUBLE}]
```

**Reply by aborruso** [reply]
Hi @pdet , I'm using v1.0.0 1f98600c2c

If I run 

```
duckdb --line -c "SELECT columns from sniff_csv('input.csv',normalize_names=true,decimal_separator=',',sample_size=-1,header=true)"
```

I have all VARCHAR

```
Columns = [{'name': cup, 'type': VARCHAR}, {'name': codice_locale_progetto, 'type': VARCHAR}, {'name': finanziamento_prosecuzione_opere_pubbliche_fpop, 'type': VARCHAR}, {'name': finanziamento_ue_diverso_da_pnrr, 'type': VARCHAR}, {'name': finanziamento_regione, 'type': VARCHAR}, {'name': finanziamento_provincia, 'type': VARCHAR}, {'name': finanziamento_comune, 'type': VARCHAR}, {'name': finanziamento_altro_pubblico, 'type': VARCHAR}, {'name': finanziamento_privato, 'type': VARCHAR}, {'name': finanziamento_da_reperire, 'type': VARCHAR}, {'name': finanziamento_pnrr, 'type': VARCHAR}, {'name': finanziamento_pnc, 'type': VARCHAR}, {'name': altri_fondi, 'type': VARCHAR}, {'name': finanziamento_totale, 'type': VARCHAR}, {'name': finanziamento_totale_pubblico, 'type': VARCHAR}, {'name': finanziamento_totale_pubblico_netto, 'type': VARCHAR}]
```

Should I then wait for the next stable release or compile duckdb myself, so I don't have this misbehavior?

Thank you

**Reply by pdet** [reply, marked as answer]
Absolutely, it does look like a bug that was corrected recently; I think the next release is currently scheduled for the first week of September!

**Reply by aborruso** [reply]
Thank you very much

**Reply by aborruso** [reply]
Wow @pdet now it works!!!!

Thank you again
